contributors/design-proposals/api-machinery/admission-control-webhooks.md

```diff
@@ -254,6 +239,24 @@ consider mechanisms to make this easier. We expect to gather user feedback
 before designing this.
 
 
+## Conversion and Versioning
+
+Webhooks will recieve the admission review subject in the exact version which
+the user sent it to the control plane. This may require the webhook to
+understand multiple versions of those types.
+
+All communication to webhooks will be JSON formatted. We will likely change that
+for GA.
+
+We need to make it possible to know whether an apiserver is safe to upgrade.
+`kube-apiserver` will set the X-Kubernetes-Version header when making any calls
```

Comment:
>The API types are typically the critical element here… e.g. "is there a field introduced in vX.Y I should be concerned with?"
>
>The versioning of the api types for an api server is orthogonal to the kubernetes machinery it was built on
>
>Both the kubernetes apimachinery (for things like objectmeta) and the apiserver (for its own types, whether it's the kube-apiserver or an extension server) versions are relevant.
>
>That said, I'm hesitant to make out-of-band content that's not in the admission review object part of the input to the admission decision. If we want schema versions (I'm not convinced we do), it should probably be part of the admission review spec

@liggitt The kubernetes release is in the binary version. We don't add fields except at release boundaries.

Reply:
If you want to set this value, seems like you'd just set it on the `AdmissionReviewSpec`.  `ServerVersion=foo`.

However, I don't think this field has value in general.  Since the originating server is often **not**, the `kube-apiserver` and doesn't know what level the `kube-apiserver` is, filling in its own value isn't likely to be the next best option.  If we need a minor version on the apiversions, I would focus on that instead of adding a product version to this API.